components/shared/net/lib.rs

```diff
@@ -528,8 +535,17 @@ pub enum CoreResourceMsg {
         IpcSender<Vec<Serde<Cookie<'static>>>>,
         CookieSource,
     ),
+    GetCookiesDataForUrlAsync(
+        CookieStoreId,
+        CookieRequestId,
+        ServoUrl,
+        String,
+        CookieSource,
+    ),
     DeleteCookies(ServoUrl),
     DeleteCookie(ServoUrl, String),
+    DeleteCookieAsync(CookieStoreId, CookieRequestId, ServoUrl, String),
+    NewCookieListener(CookieStoreId, IpcSender<CookieAsyncResponse>, ServoUrl),
```

Comment:
For lifecycle management I believe I will also need a `RemoveCookieListener` message for when a global navigates to a new page. The spec makes clear that CookieStore's relevant URL is the global's `creation_url`.

Reply:
Are you currently sending that message? I think it could be done on drop of the global.